src/query/atomWithInfiniteQuery.ts

```diff
@@ -14,28 +14,46 @@ import { queryClientAtom } from './queryClientAtom'
 import { CreateQueryOptions, GetQueryClient } from './types'
 
 export type AtomWithInfiniteQueryAction<TQueryFnData> =
-  | ({ type: 'refetch' } & Partial<
-      RefetchOptions & RefetchQueryFilters<TQueryFnData>
-    >)
+  | {
+      type: 'refetch'
+      payload: Partial<RefetchOptions & RefetchQueryFilters<TQueryFnData>>
```

Comment:
This is breaking change. We want to avoid breaking changes unless they are super necessary.
Unless `type` is conflicting, we shouldn't need to change this.

Reply:
its already conflicted with QueryTypeFilter of ReactQuery V4, maybe u can see this link for the detail
https://github.com/tannerlinsley/react-query/blob/fdbc002006d7d8c3e236da71cb0c23cf394c1835/src/core/utils.ts#L19

Reply:
Oh, I wasn't aware that. That's too bad.

Reply:
@aulneau Hey, I wonder if we can completely remove options for `type: "refetch"`. What do you think? Are there any use cases?

Reply:
hmm i'll need to dig into v4 changes of react-query a bit, but I think it could be possible to remove these and instead build out that atom we mentioned in a different thread. the atom would contain all the status/actions associated with a given query atom, rather than combining them. see here: https://github.com/pmndrs/jotai/issues/1068#issuecomment-1101737345

Reply:
This is slightly different from #1068. Here, it's our action that we defined. It's not about the returned status. I just found `useInfiniteQuery` from RQ doesn't even have `refetch` action.

Reply:
yep, u r right.